### PR TITLE
[0.17] Make validatepegin default to true if has parent chain

### PR DESCRIPTION
**Pull request by stevenroose**

### CODE REVIEW

File: src/init.cpp [hunk 1]
```diff
@@ -529,7 +529,7 @@ void SetupServerArgs()
...
 
     gArgs.AddArg("-initialfreecoins", strprintf("The amount of OP_TRUE coins created in the genesis block. Primarily for testing. (default: %d)", 0), true, OptionsCategory::DEBUG_TEST);
-    gArgs.AddArg("-validatepegin", strprintf("Validate peg-in claims. An RPC connection will be attempted to the trusted bitcoind using the `mainchain*` settings below. All functionaries must run this enabled. (default: %u)", DEFAULT_VALIDATE_PEGIN), false, OptionsCategory::ELEMENTS);
+    gArgs.AddArg("-validatepegin", "Validate peg-in claims. An RPC connection will be attempted to the trusted bitcoind using the `mainchain*` settings below. All functionaries must run this enabled. (default: true if has parent)", false, OptionsCategory::ELEMENTS);
```
Comment (instagibbs) [inline src/init.cpp]: suggested: "true if has chain has federated peg"

**Comment by instagibbs** [comment]
concept ACK

**Comment by stevenroose** [comment]
Somehow the bitcoin_functional tests think they have a parent chain. Tracking down.

File: src/init.cpp (hunk 1 above)
Comment (stevenroose) [reply src/init.cpp]: fixed

**Comment by instagibbs** [comment]
ACK

**Pull request merged**